router/core/factoryresolver.go

```diff
@@ -200,6 +209,40 @@ func (l *Loader) Load(engineConfig *nodev1.EngineConfiguration) (*plan.Configura
 					subscriptionUseSSE = *in.CustomGraphql.Subscription.UseSSE
 				}
 			}
+			// Header rewrite rules are specified by subgraph name, but data source
+			// only has routing and subscription URLs, so we must manually match them
+			var dataSourceRules []config.RequestHeaderRule
```

Comment:
Please put this logic into a separate function. `header_rule_engine.go` could be a good place.

Reply:
I've updated it, PTAL